Issue/6568 media multiselect toast

nbradbury opened the pull request:
Fixes #6568 - This PR reduces the annoyability of the toast that lets users know they can long press to multi-select in the photo picker and WP media library.

It does this by waiting until the user has used the feature three times before showing the toast, and then only showing it if we know they've never long pressed to multi-select. In addition, if we show the long press toast in the media picker, we disable showing it in the WP media library (and vice versa) since it's safe to assume if we tell the user how to multi-select in one place they can figure out how to do it in the other.

hypest commented on WordPress/src/main/java/org/wordpress/android/util/SmartToast.java:
```diff
@@ -10,51 +10,123 @@
+
+    /*
+     * prevent the passed smart toast type from being shown by setting its counter to the max - this should be
+     * used to disable long press toasts when the user long presses to multiselect since they alraady know
```
Minor: typo in `alraady`.

hypest commented on WordPress/src/main/java/org/wordpress/android/util/SmartToast.java:
```diff
@@ -10,51 +10,123 @@
+     * get the preference key which stores the number of times the feature associated with the smart toast type has
+     * been used
+     */
+    private static UndeletablePrefKey getNumTimesFeatureUsedKey(@NonNull SmartToastType type) {
```
How about introducing a field in `SmartToastType` enum itself, holding the key? That should remove repetition and remove the need to check for the `default` case. [Example implementation.](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/HelpshiftHelper.java#L42-L55)

Same for `getNumTimesToastShownKey()`.

nbradbury commented:
@hypest Good call on moving the pref key methods to the enum - I added this in e2c96f6 and fixed the typo.

When I started this PR, I did combine the two pref keys into one (see 9534791) but they really do need to be separate in order to prevent showing the toast for the WP media library when it has already been shown for the media picker (and vice versa).

hypest commented:
> prevent showing the toast for the WP media library when it has already been shown for the media picker (and vice versa)

I might be missing something but yeah, exactly for that reason is why I'm thinking a single key/counter/etc would suffice. So, the 2 cases (media library and picker) would practically share the same key/counter/etc.

hypest commented:
:shipit: !
